.github/workflows/ci-cd.yml

```diff
@@ -54,42 +54,80 @@ jobs:
+          $gitNewVersion = if ("${{ steps.tag_generator.outputs.new_version }}") {"${{ steps.tag_generator.outputs.new_version }}"} else {$null}
+          $projectCurrentVersion = ([xml](Get-Content .\src\GuardClauses\GuardClauses.csproj)).Project.PropertyGroup.Version
+          $projectNewVersion = $gitNewVersion ?? $projectCurrentVersion
+          dotnet build .\GuardClauses.sln --configuration Release -property:Version=$projectNewVersion
 
       - name: Run Unit Tests
         id: run_unit_tests
         shell: pwsh
         run: |
           dotnet test .\GuardClauses.sln --verbosity normal
 
+      - if: steps.tag_generator.outputs.new_version != ''
+        name: Upload artifacts
+        id: upload_artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: GuardClauses v${{ steps.tag_generator.outputs.new_version }}
+          path: src\GuardClauses\bin\Release\
+
   cd:
-      if: needs.ci.outputs.is_push_to_default_branch == 'true'
-      name: Continuous Deployment
-      needs: ci
-      runs-on: windows-latest
-      steps:
-        - name: Setup dotnet
-          uses: actions/setup-dotnet@v1
-          with:
-            dotnet-version: 5.0.202
+    if: needs.ci.outputs.latest_version != ''
+    name: Continuous Deployment
+    needs: ci
+    runs-on: windows-latest
+    steps:
+      - name: Setup dotnet
+        id: setup_dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 3.1.402
 
-        - name: Checkout repository
-          id: checkout_repo
-          uses: actions/checkout@v2
-          with:
-            token: ${{ secrets.GITHUB_TOKEN }}
-            fetch-depth: 0
+      - name: Checkout repository
+        id: checkout_repo
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0
 
-        - name: Publish to NuGet
-          id: publish_to_nuget
-          uses: rohith/publish-nuget@v2
-          with:
-            PROJECT_FILE_PATH: src/GuardClauses/GuardClauses.csproj
-            VERSION_REGEX: <Version>(.*)<\/Version>
-            TAG_COMMIT: true
-            TAG_FORMAT: GuardClauses-v*
-            NUGET_KEY: ${{secrets.NUGET_API_KEY}}
\ No newline at end of file
+      - name: Download and extract Artifacts
+        id: dl_artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: GuardClauses v${{ needs.ci.outputs.latest_version }}
+          path: Artifacts\
+
+      - name: Publish to NuGet
+        id: publish_to_nuget
+        uses: rohith/publish-nuget@v2
+        with:
+          PROJECT_FILE_PATH: src/GuardClauses/GuardClauses.csproj
+          VERSION_REGEX: <Version>(.*)<\/Version>
+          TAG_COMMIT: true
+          TAG_FORMAT: GuardClauses-v*
+          NUGET_KEY: ${{secrets.NUGET_API_KEY}}
+
+      - name: Create and publish release
+        id: create_release
+        uses: softprops/action-gh-release@v1
```

Comment:
Consider adding more attributes if any could be used to reproduce the way the owner manually submits their release. We will discuss this tomorrow.

Reply:
When you add a comment under the prefix for the release the Release is taking the commit message and publishing it like the owner is publishing his release.